src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,169 @@
+package edu.hm.hafner.util;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.*;
+import java.util.HashSet;
+
+/**
+ * Tests the class HashSet.
+ *
+ * @author Sofia Steinleitner
+ */
+class SetTest {
+    private static final Integer eins = 1;
+    private static final Integer zwei = 2;
+    private static final Integer drei = 3;
+    private static final Integer vier = 4;
+
+    @Test
+    void shouldTestConstructorHashSet() {
+        HashSet<Integer> set = new HashSet<Integer>();
+        Assertions.assertThat(set).isEmpty();
+    }
+
+    @Test
+    void shouldTestConstructorCollection() {
+        Assertions.assertThatThrownBy(() -> new HashSet<Integer>(null))
+                .isExactlyInstanceOf(NullPointerException.class)
+                .as("If the specified collection is null, throws NullPointerException");
+    }
+
+    @Test
+    void shouldTestConstructorCapacity() {
+        Assertions.assertThatThrownBy(() -> new HashSet<Integer>(-1))
+                .isExactlyInstanceOf(IllegalArgumentException.class)
+                .as("If capacity less than zero, throws IllegalArgumentException");
+    }
+
+    @Test
+    void shouldTestConstructorCapacityLoadFactor() {
+        Assertions.assertThatThrownBy(() -> new HashSet<Integer>(-1, -15))
+                .isExactlyInstanceOf(IllegalArgumentException.class)
+                .as("If capacity less than zero or loadFactor nonpositve,"
+                        + " throws IllegalArgumentException");
+
+        Assertions.assertThatThrownBy(() -> new HashSet<Integer>(1, -15))
+                .isExactlyInstanceOf(IllegalArgumentException.class)
+                .as("If capacity less than zero or loadFactor nonpositve,"
+                        + " throws IllegalArgumentException");
+
+        Assertions.assertThatThrownBy(() -> new HashSet<Integer>(-1, 15))
+                .isExactlyInstanceOf(IllegalArgumentException.class)
+                .as("If capacity less than zero or loadFactor nonpositve,"
+                        + " throws IllegalArgumentException");
+    }
+
+    @Test
+    void shouldAddElements() {
+        HashSet<Integer> set = new HashSet<Integer>();
+        set.add(eins);
+        set.add(zwei);
+
+        Assertions.assertThat(set.add(drei)).isTrue();
+        Assertions.assertThat(set.add(zwei)).isFalse();
+    }
+
+    @Test
+    void shouldContainElements() {
+        HashSet<Integer> set = new HashSet<Integer>();
+        set.add(eins);
+        set.add(zwei);
+
+        Assertions.assertThat(set).contains(eins, zwei);
+        Assertions.assertThat(set).doesNotContain(vier);
```

Comment:
```suggestion
        Assertions.assertThat(set.contains(eins)).isTrue();
        Assertions.assertThat(set.contains(zwei)).isTrue();
        Assertions.assertThat(set.contains(vier)).isFalse();
```
Wenn ein Test eine Methode testen soll, wie hier `HashSet#contains`, dann im Test auch diese explizit aufrufen.